src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -161,83 +165,95 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
   }
 
   function commitAllWork(finishedWork : Fiber) {
-    // Commit all the side-effects within a tree.
-    // First, we'll perform all the host insertions, updates, deletions and
-    // ref unmounts.
-    let effectfulFiber = finishedWork.firstEffect;
-    while (effectfulFiber) {
-      switch (effectfulFiber.effectTag) {
-        case Placement:
-        case PlacementAndCallback: {
-          commitInsertion(effectfulFiber);
-          // Clear the "placement" from effect tag so that we know that this is inserted, before
-          // any life-cycles like componentDidMount gets called.
-          effectfulFiber.effectTag ^= Placement;
-          break;
+    if (!isCommitting) {
+      isCommitting = true;
+      config.prepareForCommit();
```

Comment:
These should both be imported as constants like these https://github.com/facebook/react/blob/e3131c1d55d6695c2f0966379535f88b813f912b/src/renderers/shared/fiber/ReactFiberCommitWork.js#L35-L41